constraint-solver/src/solver.rs

```diff
@@ -21,11 +24,17 @@ mod quadratic_equivalences;
 
 /// Solve a constraint system, i.e. derive assignments for variables in the system.
 pub fn solve_system<T, V>(
-    constraint_system: ConstraintSystem<T, V>,
-    bus_interaction_handler: impl BusInteractionHandler<T>,
+    constraint_system: ConstraintSystemGeneric<T, V>,
+    bus_interaction_handler: impl BusInteractionHandler<<T::Transformed as RuntimeConstant>::FieldType>,
 ) -> Result<SolveResult<T, V>, Error>
 where
-    T: FieldElement,
+    T: RuntimeConstant + VarTransformable<V, Variable<V>> + Display,
+    T::Transformed: RuntimeConstant
+        + VarTransformable<Variable<V>, V, Transformed = T>
```

Comment:
Right, so this is saying that we should be able to transform back. We'd expect this for all `T: VarTransformable<V1, V2>`, right? Maybe we can add this bound to `VarTransformable::Transformed` already?